zulipterminal/ui_tools/boxes.py

```diff
@@ -219,6 +219,47 @@ def private_header(self) -> Any:
         header.markup = title_markup
         return header
 
+    def top_header_bar(self, message_view: Any) -> Any:
+        if self.message['type'] == 'stream':
+            return message_view.stream_header()
+        else:
+            return message_view.private_header()
+
+    def top_search_bar(self) -> Any:
+        if self.model.narrow == []:
+            text_to_fill = 'All messages'
+        elif len(self.model.narrow) == 1 and \
+                self.model.narrow[0][1] == 'private':
+            text_to_fill = 'All private messages'
+        elif len(self.model.narrow) == 1 and \
+                self.model.narrow[0][1] == 'starred':
+            text_to_fill = 'Starred messages'
+        elif self.message['type'] == 'stream':
+            bar_color = self.model.stream_dict[self.stream_id]['color']
+            bar_color = 's' + bar_color[:2] + bar_color[3] + bar_color[5]
+            if len(self.model.narrow) == 2 and \
+                    self.model.narrow[1][0] == 'topic':
+                text_to_fill = ('bar', [  # type: ignore
+                    (bar_color, '{}'.format(self.caption)),
+                    (bar_color, ': topic narrow ')
```

Comment:
I'm not too keen on this, though unless we collapse the bar below I'm not sure what to do other than use a better phrase.

However, there's an extra space at the end here? That would probably be clearer if the tests were more complete?

Reply:
We could go for a better phrase. :)
Will amend tests. I will definitely do a v2 of this shortly, - merge the two bars in smaller narrows.